spdmlib/src/requester/encap_req.rs

```diff
@@ -0,0 +1,212 @@
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+
+use codec::{Codec, Reader, Writer};
+
+use crate::{
+    common::SpdmCodec,
+    config,
+    error::{
+        SpdmResult, SPDM_STATUS_INVALID_MSG_FIELD, SPDM_STATUS_INVALID_MSG_SIZE,
+        SPDM_STATUS_UNSUPPORTED_CAP,
+    },
+    message::{
+        SpdmDeliverEncapsulatedResponsePayload, SpdmEncapsulatedRequestPayload,
+        SpdmEncapsulatedResponseAckPayload, SpdmEncapsulatedResponseAckPayloadType, SpdmErrorCode,
+        SpdmGetDigestsRequestPayload, SpdmGetEncapsulatedRequestPayload,
+        SpdmKeyExchangeMutAuthAttributes, SpdmMessage, SpdmMessageHeader, SpdmMessagePayload,
+        SpdmRequestResponseCode, ENCAPSULATED_RESPONSE_ACK_HEADER_SIZE,
+    },
+    protocol::{SpdmRequestCapabilityFlags, SpdmResponseCapabilityFlags, SpdmVersion},
+};
+
+use super::RequesterContext;
+
+impl<'a> RequesterContext<'a> {
+    pub fn get_encapsulated_request_response(
+        &mut self,
+        session_id: u32,
+        mut_auth_requested: SpdmKeyExchangeMutAuthAttributes,
+    ) -> SpdmResult {
+        if self.common.negotiate_info.spdm_version_sel.get_u8()
+            < SpdmVersion::SpdmVersion11.get_u8()
+        {
+            return Err(SPDM_STATUS_UNSUPPORTED_CAP);
+        }
+
+        if !self
+            .common
+            .negotiate_info
+            .req_capabilities_sel
+            .contains(SpdmRequestCapabilityFlags::ENCAP_CAP)
+            || !self
+                .common
+                .negotiate_info
+                .rsp_capabilities_sel
+                .contains(SpdmResponseCapabilityFlags::ENCAP_CAP)
+        {
+            return Err(SPDM_STATUS_UNSUPPORTED_CAP);
+        }
+
+        match mut_auth_requested {
+            // Optimized session-based mutual authentication
+            // When the Requester successfully receives a Session-Secrets-Exchange response with an included encapsulated
+            // request (GET_DIGEST), the Requester shall send a DELIVER_ENCAPSULATED_RESPONSE after processing the encapsulated request.
+            SpdmKeyExchangeMutAuthAttributes::MUT_AUTH_REQ_WITH_GET_DIGESTS => {
+                let mut encapsulated_request = [0u8; 4];
+                let mut writer = Writer::init(&mut encapsulated_request);
+                let get_digest_request = SpdmMessage {
+                    header: SpdmMessageHeader {
+                        version: self.common.negotiate_info.spdm_version_sel,
+                        request_response_code: SpdmRequestResponseCode::SpdmRequestGetDigests,
+                    },
+                    payload: SpdmMessagePayload::SpdmGetDigestsRequest(
+                        SpdmGetDigestsRequestPayload {},
+                    ),
+                };
+                let _ = get_digest_request.spdm_encode(&mut self.common, &mut writer)?;
+                self.process_encapsulated_request(session_id, 0, &encapsulated_request)?;
+            }
+            _ => {
```

Comment:
I think we need change this to `MUT_AUTH_REQ_WITH_ENCAP_REQUEST`.

The `_` should be treated as error.